tests/integration/test_charm.py

```diff
@@ -5,19 +5,104 @@
 import logging
 
 import pytest
+from helpers import (
+    check_prometheus_is_ready,
+    get_job_config_for,
+    get_prometheus_config,
+    get_prometheus_rules,
+    get_rules_for,
+)
 
 logger = logging.getLogger(__name__)
 
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy_with_ubuntu_image(ops_test):
+@pytest.mark.parametrize("oci_image", ["ubuntu/prometheus:latest", "prom/prometheus:latest"])
+async def test_build_and_deploy_with_alternative_images(ops_test, prometheus_charm, oci_image):
     """Test that the Prometheus charm can be deployed successfully."""
-    prometheus_charm = await ops_test.build_charm(".")
-    resources = {"prometheus-image": "ubuntu/prometheus:latest"}
+    resources = {"prometheus-image": oci_image}
+    app_name = "prometheus-" + oci_image.split("/")[0]
 
+    await ops_test.model.deploy(prometheus_charm, resources=resources, application_name=app_name)
+    await ops_test.model.wait_for_idle(apps=[app_name], status="active")
+    await ops_test.model.block_until(lambda: len(ops_test.model.applications[app_name].units) > 0)
+
+    assert ops_test.model.applications[app_name].units[0].workload_status == "active"
+
+    await check_prometheus_is_ready(ops_test, app_name, 0)
+
+    await ops_test.model.applications[app_name].remove()
+    await ops_test.model.reset()
```

Comment:
Why not carry forward the built app?

Reply:
I want each test to be independent of the other.

Reply:
This **might** be an issue in case a test fails prior to reset being called. Instead, I would suggest adding a fixture with the `@pytest.fixture(autouse=True)` decorator, which will run the reset prior to every test case.

Reply:
To my understanding, "model reset" should not be used in an integration test: after all, it's an integration test: the app should be able to handle whatever happens in whatever sequence. I think that "reset" is used here instead of creating another integration test file (which has the unpleasant overhead of rebuilding and redeploying everything).

Reply:
I have pointed this out in response to other comments of yours. No charm is built more than once. The redeployment is intentional. The point is that each test can be executed by itself. Not only is it good practice that tests are independent of each other but also it allows executing just a single test if so desired, for example during development. The Model reset ensures that there are no lingering artifacts from a previous test, which can happen if the previous test failed and did not clean up properly.

Reply:
> This **might** become an issue in case a test fails prior to reset being called, but since you've marked all of them with `abort_on_fail` that will never make any difference. However, I would suggest instead adding a fixture with the `@pytest.fixture(autouse=True)` decorator, which will run the reset prior to every test case. No need to act on it in this PR though, but could be nice to do in a future refactoring.

OK. will explore if `autouse` can be used so that it is executed at end of test rather. You make a good point regarding `abort_on_fail`. I think this could be removed now.  I would like to remove it if you agree. Or perhaps we can choose to selectively mark tests as abort on fail, for example only if the first two tests (deployment check of two charms) fail then abort on fail. I can pick this up in next PR after discussion in plenary when I am back.

Reply:
Add to the TODO list ticket https://github.com/canonical/prometheus-operator/issues/156